client/main.lua

```diff
@@ -99,329 +85,350 @@ function AlertPolice()
-                end
-
-            else
-                Citizen.Wait(500)
-            end
-
-            if dist <= 7 and BlownUp == 0 and PlayerJob.name ~= 'police' then
-                if BlowBackdoor == 0 then
-                    hintToDisplay('Press [G] to blow up the back door and take the money')
-                    BlowBackdoor = 1
-                end
-                if IsControlPressed(0, 47) and GuardsDead == 1 then
-                    CheckVehicleInformation()
-                    TriggerEvent("qb-armoredtruckheist:client:911alert")
-                    Citizen.Wait(500)
-                end
-            end
+function stopAndBeAngry()
+	Citizen.CreateThread(function()
+		SetVehicleBrake(transport)
+		Wait(1000)
+
+		GiveWeaponToPed(navigator, Config.NavWeap, 420, 0, 1)
+		GiveWeaponToPed(navigator2, Config.NavWeap, 420, 0, 1)
+		GiveWeaponToPed(pilot, Config.DriverWeap, 420, 0, 1)
+
+		SetPedDropsWeaponsWhenDead(navigator,false)
+		SetPedRelationshipGroupDefaultHash(navigator,`COP`)
+		SetPedRelationshipGroupHash(navigator,`COP`)
+		SetPedAsCop(navigator,true)
+		SetCanAttackFriendly(navigator,false,true)
+
+		SetPedDropsWeaponsWhenDead(navigator2,false)
+		SetPedRelationshipGroupDefaultHash(navigator2,`COP`)
+		SetPedRelationshipGroupHash(navigator2,`COP`)
+		SetPedAsCop(navigator2,true)
+		SetCanAttackFriendly(navigator2,false,true)
+
+		SetPedDropsWeaponsWhenDead(pilot,false)
+		SetPedRelationshipGroupDefaultHash(pilot,`COP`)
+		SetPedRelationshipGroupHash(pilot,`COP`)
+		SetPedAsCop(pilot,true)
+		SetCanAttackFriendly(pilot,false,true)
+
+		TaskCombatPed(pilot, GetPlayerPed(-1), 0, 16)
+		TaskCombatPed(navigator, GetPlayerPed(-1), 0, 16)
+		TaskCombatPed(navigator2, GetPlayerPed(-1), 0, 16)
+
+		TaskEveryoneLeaveVehicle(transport)
+	end)
+end
 
-        else
-            Citizen.Wait(1500)
-        end
-    end
-end)
+--Crims side of the mission
+function startMission()
+	Citizen.CreateThread(function()
+		while MissionStart == 1 do
+			local plyCoords = GetEntityCoords(PlayerPedId(), false)
+			local transCoords = GetEntityCoords(transport)
+			local dist = #(plyCoords - transCoords)
+			if dist <= 55.0  then
+				DrawMarker(0, transCoords.x, transCoords.y, transCoords.z+4.5, 0, 0, 0, 0, 0, 0, 1.0, 1.0, 1.0, 135, 31, 35, 100, 1, 0, 0, 0)
+				if warning == 0 then
+					warning = 1
+					stopAndBeAngry()
+					QBCore.Functions.Notify("Get rid of the guards before you place the bomb.", "error")
+					AlertPolice()
+				end
+
+				if GuardsDead == 0 then
+					CheckGuards()
+				end
+
+				if dist <= 7 and BlownUp == 0 and PlayerJob.name ~= 'police' then
+					if BlowBackdoor == 0 then
+						hintToDisplay('Press [G] to blow up the back door and take the money')
+						if IsControlJustPressed(0, 47) then
+							BlowBackdoor = 1
+							CheckVehicleInformation()
+							TriggerEvent("qb-armoredtruckheist:client:911alert")
+							hideLastHint()
+							Wait(5000)
+						end
+					end
+				end
+			end
+			Wait(0)
```

Comment:
This one can have a sleep of 500 as well before turning to 0 after the dist is under 55